funcx_endpoint/funcx_endpoint/executors/high_throughput/executor.py

```diff
@@ -148,6 +148,14 @@ class HighThroughputExecutor(StatusHandlingExecutor, RepresentationMixin):
     mem_per_worker : float
         Memory to be assigned to each worker. Default=None(no limits)
 
+    available_accelerators: int, list of str
+        Either a list of accelerators device IDs
+        or an integer defining the number of accelerators available.
+        If an integer, sequential device IDs will be created starting at 0.
+        The manager will ensure each worker is pinned to an accelerator and
```

Comment:
The specific pinning mechanism here is setting environment variables like this: `environment_variables["CUDA_VISIBLE_DEVICES"] = device` in the worker, right?

Reply:
When I'm configuring this as an endpoint administrator, I have to pay attention to the number of workers that the endpoint will try to start, compared to the available accelerators, right?

Looks like a ValueError will be raised later on if the code tries to start too many workers compared to the number of available_accelerators.

If this is true, maybe the number of accelerators should be included in the max_worker_count calculation:

```
        self.max_worker_count = min(
            max_workers, math.floor(self.cores_on_node / cores_per_worker)
        )
```

which takes into account other constraints on maximum workers - if the behaviour is to only start as many workers as there are cores for, even if I said max_workers=99999999, then I think the behaviour should also be to only start as many workers as there are accelerators for, even if I said max_workers=9999999.

Reply:
Done